apollo-router/src/plugins/connectors/make_requests.rs

```diff
@@ -573,12 +653,74 @@ fn entities_with_fields_from_request(
         .collect::<Result<Vec<_>, _>>()
 }
 
+// --- BATCH ENTITIES ----------------------------------------------------------------
+
+/// Connectors on types can make a single batch request for multiple entities
+/// using the `$batch` variable.
+///
+/// The key (pun intended) to batching is that we have to return entities in an
+/// order than matches the `representations` variable. We use the "key" fields
+/// to determine
+fn batch_entities_from_request(
+    connector: Arc<Connector>,
+    request: &connect::Request,
+) -> Result<Vec<ResponseKey>, MakeRequestError> {
+    use MakeRequestError::*;
+
+    let Some(keys) = &request.keys else {
+        return Err(InvalidOperation("TODO better error type".into()));
+    };
+
+    let Some(representations) = request.variables.variables.get(REPRESENTATIONS_VAR) else {
+        unreachable!("batch_entities_from_request called without representations");
```

Comment:
```suggestion
        return Err(InvalidRepresentations("Internal error".into()));
```

Even if we think it's unreachable, let's not panic 😁